src/neptune/cli/sync.py

```diff
@@ -50,60 +51,122 @@
 
 class SyncRunner:
     @staticmethod
-    def sync_all_offline(*, backend: "NeptuneBackend", base_path: Path, project_name: Optional[str] = None) -> None:
+    def sync_all_offline(
+        *,
+        backend: "NeptuneBackend",
+        base_path: Path,
+        project_name: Optional[str] = None,
+        num_threads: Optional[int] = None,
+    ) -> None:
         containers = collect_containers(path=base_path, backend=backend)
 
-        project = get_project(project_name_flag=QualifiedName(project_name) if project_name else None, backend=backend)
+        project = get_project(
+            project_name_flag=QualifiedName(project_name) if project_name else None,
+            backend=backend,
+        )
         if not project:
             raise CannotSynchronizeOfflineRunsWithoutProject
 
-        for container in containers.offline_containers:
-            container.sync(base_path=base_path, backend=backend, project=project)
+        with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
```

Comment:
What if `num_threads` is `None`? I think we should perform an ordinary loop then, right?

Reply:
`ThreadPoolExecutor` uses [it's defaults](https://getdocs.org/Python/docs/3.10/library/concurrent.futures#:~:text=Changed%20in%20version%203.8%3A%20Default%20value%20of%20max_workers,for%20CPU%20bound%20tasks%20which%20release%20the%20GIL.) if `num_workers` is `None`

Reply:
So what if a user doesn't want to use threads?

Reply:
`neptune sync -n 1` ;)

Reply:
But it's anyway a design choice that the team should discuss internally. Save for that and the other points LGTM ;)